src/site/content/en/blog/streams/index.md

```diff
@@ -0,0 +1,966 @@
+  async start(controller) {
+    // Called by constructor.
+    console.log('[start]');
+    controller.enqueue('a');
+    controller.enqueue('b');
+    controller.enqueue('c');
+  },
+  async pull(controller) {
+    // Called read when controller's queue is empty.
+    console.log('[pull]');
+    controller.enqueue('d');
+    controller.close();
+  },
+  async cancel(reason) {
+    // Called when the stream is canceled.
+    console.log('[cancel]', reason);
+  },
+});
+
+(async () => {
+  // Create two `ReadableStream`s.
+  const [streamA, streamB] = readableStream.tee();
+
+  // Read streamA iteratively one by one.
+  const readerA = streamA.getReader();
+  console.log('[A]', await readerA.read()); //=> {value: "a", done: false}
+  console.log('[A]', await readerA.read()); //=> {value: "b", done: false}
+  console.log('[A]', await readerA.read()); //=> {value: "c", done: false}
+  console.log('[A]', await readerA.read()); //=> {value: "d", done: false}
+  console.log('[A]', await readerA.read()); //=> {value: undefined, done: true}
+
+  // Read streamB in a loop.
+  const readerB = streamB.getReader();
+  for (let result = await readerB.read(); !result.done; result = await readerB.read()) {
+    console.log('[B]', result);
+  }
+})();
+```
+
+## Readable byte streams
+
+For streams representing bytes, an extended version of the readable stream is provided to handle
+bytes efficiently, in particular by minimizing copies. Byte streams allow for bring-your-own-buffer
+(BYOB) readers to be acquired. The default implementation can give a range of different outputs such
+as strings or array buffers in the case of WebSockets, whereas byte streams guarantee byte output.
+In addition to that, being able to have BYOB readers has benefits in terms of stability. This is
+because if a buffer detaches, it can guarantee that one does not write into the same buffer twice,
+hence avoiding race conditions. BYOB readers can reduce the number of times the browser needs to run
+garbage collection, because it can reuse buffers.
+
+You can create a readable byte stream by passing an additional `type` parameter to the
+`ReadableStream()` constructor.
+
+```js
+new ReadableStream({ type: 'bytes' });
+```
+
+You can then get access to a `ReadableStreamBYOBReader` by setting the `mode` parameter accordingly:
+`ReadableStream.getReader({ mode: "byob" })`. This allows for more precise control over buffer
+allocation in order to avoid copies. To read from the byte stream, you need to call
+`ReadableStreamBYOBReader.read(view)`, where `view` is an
+[`ArrayBufferView`](https://developer.mozilla.org/en-US/docs/Web/API/ArrayBufferView).
+
+```js
+const reader = readableStream.getReader({ mode: 'byob' });
+
+let startingAB = new ArrayBuffer(1_024);
+const buffer = await readInto(startingAB);
+console.log('The first 1,024 bytes: ', buffer);
+
+async function readInto(buffer) {
+  let offset = 0;
+
+  while (offset < buffer.byteLength) {
```

Comment:
Ah, so it isn't required to fill the view you give it? I thought it might fill the view aside from the final chunk.

Reply:
I don't know the answer. This example comes from the explainer.